### PR TITLE
Oxedions/feat/nmcli

### PR DESCRIPTION
**Pull request by oxedions**
Adapt nic nmcli role to new inventory format.

WARNING: this role needs **Ansible 2.10** to work properly.

### CODE REVIEW

**Comment by oxedions** [comment]
This role is expected to replace current nic role for 1.4 version.
However, I propose to merge it at is for now, and add advanced features later (still before 1.4).

**Comment by btravouillon** [comment]
> This role is expected to replace current nic role for 1.4 version.

Will it move to core/nic in 1.4?

**Comment by oxedions** [comment]
This would be the target yes. But only if at least all features of current nic are added 🙂

File: roles/addons/nic_nmcli/tasks/main.yml [hunk 1]
```diff
@@ -19,79 +19,54 @@
...
+#    path: /etc/NetworkManager/conf.d/10-globally-managed-devices.conf
+#    state: touch
+#  when:
+#    - ansible_facts.distribution_major_version == "18"
+#    - ansible_facts.os_family == "Ubuntu"
 
 # Note from JK
 # We could add so much more features here, like VLAN, LACP, MTU, etc. Easy to add.
-- name: Set NIC configuration
+- name: nmcli █ Set NIC configuration using nmcli
```
Comment (btravouillon) [inline roles/addons/nic_nmcli/tasks/main.yml:46]: Isn't it redundant?

File: roles/addons/nic_nmcli/tasks/main.yml [hunk 2]
```diff
@@ -19,79 +19,54 @@
...
   nmcli:
-    conn_name: "{{ item|string }}"
-    type: "{% if network_interfaces[item].type is defined and not none %}{{ network_interfaces[item].type }}{% else %}ethernet{% endif %}"
-#    ifname: "{{ network_connections[item].ifname }}"
-    ip4: "{{ network_interfaces[item].ip4 }}/{{ networks[network_interfaces[item].network].prefix }}"
+    conn_name: "{{ item.name | string }}"
+    type: "{% if item.type is defined and item.type is not none %}{{ item.type }}{% else %}ethernet{% endif %}"
+#    ifname: "{{ network_connectionsitem.ifname }}"
```
Comment (btravouillon) [inline roles/addons/nic_nmcli/tasks/main.yml:50]: Remove ?

**Review by btravouillon — COMMENTED** [review]
One can also use this module with ansible 2.9 + [community.general.nmcli](https://docs.ansible.com/ansible/latest/collections/community/general/nmcli_module.html).

File: roles/addons/nic_nmcli/tasks/main.yml (hunk 1 above)
Comment (oxedions) [reply roles/addons/nic_nmcli/tasks/main.yml:46]: yes, you are right. I will correct this.

File: roles/addons/nic_nmcli/tasks/main.yml (hunk 2 above)
Comment (oxedions) [reply roles/addons/nic_nmcli/tasks/main.yml:50]: yes !

**Comment by oxedions** [comment]
> ansible 2.9 + community.general.nmcli.
You are right. I used it this way 🙂

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**